components/tikv_util/src/security.rs

```diff
@@ -166,6 +173,68 @@ impl ServerCredentialsFetcher for Fetcher {
     }
 }
 
+/// Check peer CN with cert-allowed-cn field.
+/// Return true when the match is successful (support wildcard pattern).
+/// Skip the check when cert-allowed-cn is not set or the secure channel is not used.
+pub fn check_common_name(cert_allowed_cn: &HashSet<String>, ctx: &RpcContext) -> bool {
+    if cert_allowed_cn.is_empty() {
+        return true;
+    }
+    if let Some(auth_ctx) = ctx.auth_context() {
+        if let Some(auth_property) = auth_ctx
+            .into_iter()
+            .find(|x| x.name() == "x509_common_name")
+        {
+            let peer_cn = auth_property.value_str().unwrap();
+            match_peer_names(cert_allowed_cn, peer_cn)
+        } else {
+            false
+        }
+    } else {
+        true
+    }
+}
+
+fn match_peer_names(allowed_cn: &HashSet<String>, name: &str) -> bool {
+    for cn in allowed_cn {
+        if do_match_name(cn, name) {
+            return true;
+        }
+    }
+    false
+}
+
+fn do_match_name(pattern: &str, name: &str) -> bool {
+    if pattern.is_empty() {
+        return false;
+    }
+    // Try exact match
+    if pattern == name {
```

Comment:
"." == "." should be an invalid match.

Reply:
@tennix Can you provide a specific match rule